File: leetcode/lc-us/cpp/205.cpp
#include <string>

using namespace std;

class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int ms[128] = {};
        int mt[128] = {};
        for (int i = 0; i < s.length(); i++) {
            if (ms[s[i]] && ms[s[i]] != t[i] || mt[t[i]] && mt[t[i]] != s[i]) return false;
            ms[s[i]] = t[i];
            mt[t[i]] = s[i];
        }
        return true;
    }
};

File: leetcode/lc-us/rust/205.rs
use std::collections::HashMap;

impl Solution {
    pub fn is_isomorphic(s: String, t: String) -> bool {
        let mut st: HashMap<char, char> = HashMap::new();
        let mut ts: HashMap<char, char> = HashMap::new();
        for (a, b) in s.chars().zip(t.chars()) {
            if let Some(c) = st.insert(a, b) {
                if b != c {
                    return false;
                }
            }
            if let Some(c) = ts.insert(b, a) {
                if a != c {
                    return false;
                }
            }
        }
        true
    }
}